### ISSUE TITLE
Blin 2020.10, round 1

### ISSUE THREAD

**Issue by Altai-man** [rakudo/rakudo#3966]
[Blin](https://github.com/Raku/Blin) results between 2020.09 ([41b5c08](https://github.com/rakudo/rakudo/commit/41b5c0815db394e874296b4d81841d227b568cad)) and HEAD ([1971ba4](https://github.com/rakudo/rakudo/commit/1971ba4642057e8b1b723f9a257308a0069b48af)):

* [ ] [AI::FANN](https://modules.raku.org/dist/AI::FANN) – Fail, Bisected: [e0e24a8](https://github.com/rakudo/rakudo/commit/e0e24a8dbfd73b20697096467f499f8def5cea3a)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: AI::FANN
  ===> Found: AI::FANN:ver<0.0.1>:auth<github:perlpilot> [via Zef::Repository::LocalCache]
  ===> Testing: AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  [AI::FANN] t/00-creation.t .. ok
  [AI::FANN] All tests successful.
  [AI::FANN] Files=1, Tests=14, 36 wallclock secs ( 0.02 usr  0.01 sys + 22.44 cusr  1.35 csys = 23.82 CPU)
  [AI::FANN] Result: PASS
  ===> Testing [OK] for AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  ===> Installing: AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  ===> Install [OK] for AI::FANN:ver<0.0.1>:auth<github:perlpilot>

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: AI::FANN
  ===> Found: AI::FANN:ver<0.0.1>:auth<github:perlpilot> [via Zef::Repository::LocalCache]
  ===> Testing: AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  [AI::FANN] t/00-creation.t .. 
  [AI::FANN] All 10 subtests passed 
  [AI::FANN] Test Summary Report
  [AI::FANN] -------------------
  [AI::FANN] t/00-creation.t (Wstat: 139 Tests: 10 Failed: 0)
  [AI::FANN]   Non-zero wait status: 139
  [AI::FANN]   Parse errors: No plan found in TAP output
  [AI::FANN] Files=1, Tests=10, 29 wallclock secs ( 0.01 usr  0.00 sys + 19.51 cusr  1.20 csys = 20.72 CPU)
  [AI::FANN] Result: FAIL
  ===> Testing [FAIL]: AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  [AI::FANN] Failed to get passing tests, but continuing with --force-test
  ===> Installing: AI::FANN:ver<0.0.1>:auth<github:perlpilot>
  ===> Install [OK] for AI::FANN:ver<0.0.1>:auth<github:perlpilot>

  ```
  </details>
* [ ] [Compress::Bzip2](https://modules.raku.org/dist/Compress::Bzip2) – Fail, Bisected: [ecaf334](https://github.com/rakudo/rakudo/commit/ecaf334c851de35c6e8fbca6a4012bcc4d21246c)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: Compress::Bzip2
  ===> Found: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man> [via Zef::Repository::LocalCache]
  ===> Testing: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  [Compress::Bzip2] t/10-basic.t .. ok
  [Compress::Bzip2] All tests successful.
  [Compress::Bzip2] Files=1, Tests=10,  8 wallclock secs ( 0.02 usr  0.00 sys +  5.44 cusr  0.43 csys =  5.89 CPU)
  [Compress::Bzip2] Result: PASS
  ===> Testing [OK] for Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  ===> Installing: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  ===> Install [OK] for Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: Compress::Bzip2
  ===> Found: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man> [via Zef::Repository::LocalCache]
  ===> Testing: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  [Compress::Bzip2] t/10-basic.t .. 
  [Compress::Bzip2] All 3 subtests passed 
  [Compress::Bzip2] Test Summary Report
  [Compress::Bzip2] -------------------
  [Compress::Bzip2] t/10-basic.t (Wstat: 139 Tests: 3 Failed: 0)
  [Compress::Bzip2]   Non-zero wait status: 139
  [Compress::Bzip2]   Parse errors: No plan found in TAP output
  [Compress::Bzip2] Files=1, Tests=3,  4 wallclock secs ( 0.02 usr  0.00 sys +  2.26 cusr  0.31 csys =  2.59 CPU)
  [Compress::Bzip2] Result: FAIL
  ===> Testing [FAIL]: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  [Compress::Bzip2] Failed to get passing tests, but continuing with --force-test
  ===> Installing: Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>
  ===> Install [OK] for Compress::Bzip2:ver<0.4.1>:auth<github:Altai-man>

  ```
  </details>
* [ ] [MagickWand](https://modules.raku.org/dist/MagickWand) – Fail, Bisected: [84ce97d](https://github.com/rakudo/rakudo/commit/84ce97d255640f68e7f6300685ee0ab1c481ce2c)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: MagickWand
  ===> Found: MagickWand:ver<0.1.0>:auth<github:azawawi> [via Zef::Repository::LocalCache]
  ===> Testing: MagickWand:ver<0.1.0>:auth<github:azawawi>
  [MagickWand] t/01-load.t ......... ok
  [MagickWand] t/02-magickwand.t ... ok
  [MagickWand] t/03-read-buffer.t .. ok
  [MagickWand] All tests successful.
  [MagickWand] Files=3, Tests=11, 121 wallclock secs ( 0.02 usr  0.00 sys + 89.06 cusr  2.96 csys = 92.04 CPU)
  [MagickWand] Result: PASS
  ===> Testing [OK] for MagickWand:ver<0.1.0>:auth<github:azawawi>
  ===> Installing: MagickWand:ver<0.1.0>:auth<github:azawawi>
  ===> Install [OK] for MagickWand:ver<0.1.0>:auth<github:azawawi>

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: MagickWand
  ===SORRY!===
  Object does not exist in serialization context

  ===> Found: MagickWand:ver<0.1.0>:auth<github:azawawi> [via Zef::Repository::LocalCache]
  ===> Testing: MagickWand:ver<0.1.0>:auth<github:azawawi>
  [MagickWand] [31m===[0mSORRY![31m===[0m Error while compiling /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/01-load.t
  [MagickWand] ===SORRY!===
  [MagickWand] Object does not exist in serialization context
  [MagickWand] at /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/01-load.t:8
  [MagickWand] t/01-load.t ......... 
  [MagickWand] Dubious, test returned 1 (wstat 256, 0x100)
  [MagickWand] No subtests run 
  [MagickWand] [31m===[0mSORRY![31m===[0m Error while compiling /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/02-magickwand.t
  [MagickWand] ===SORRY!===
  [MagickWand] Object does not exist in serialization context
  [MagickWand] at /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/02-magickwand.t:8
  [MagickWand] t/02-magickwand.t ... 
  [MagickWand] Dubious, test returned 1 (wstat 256, 0x100)
  [MagickWand] No subtests run 
  [MagickWand] [31m===[0mSORRY![31m===[0m Error while compiling /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/03-read-buffer.t
  [MagickWand] ===SORRY!===
  [MagickWand] Object does not exist in serialization context
  [MagickWand] at /mnt/data/zef-data/store/MagickWand-0.1.0.tar.gz/MagickWand-0.1.0/t/03-read-buffer.t:9
  [MagickWand] t/03-read-buffer.t .. 
  [MagickWand] Dubious, test returned 1 (wstat 256, 0x100)
  [MagickWand] No subtests run 
  [MagickWand] Test Summary Report
  [MagickWand] -------------------
  [MagickWand] t/01-load.t       (Wstat: 256 Tests: 0 Failed: 0)
  [MagickWand]   Non-zero exit status: 1
  [MagickWand]   Parse errors: No plan found in TAP output
  [MagickWand] t/02-magickwand.t (Wstat: 256 Tests: 0 Failed: 0)
  [MagickWand]   Non-zero exit status: 1
  [MagickWand]   Parse errors: No plan found in TAP output
  [MagickWand] t/03-read-buffer.t (Wstat: 256 Tests: 0 Failed: 0)
  [MagickWand]   Non-zero exit status: 1
  [MagickWand]   Parse errors: No plan found in TAP output
  [MagickWand] Files=3, Tests=0, 121 wallclock secs ( 0.02 usr  0.00 sys + 91.21 cusr  3.29 csys = 94.52 CPU)
  [MagickWand] Result: FAIL
  ===> Testing [FAIL]: MagickWand:ver<0.1.0>:auth<github:azawawi>
  [MagickWand] Failed to get passing tests, but continuing with --force-test
  ===> Installing: MagickWand:ver<0.1.0>:auth<github:azawawi>
  ===> Install [FAIL] for MagickWand:ver<0.1.0>:auth<github:azawawi>: ===SORRY!===
  Object does not exist in serialization context


  ```
  </details>
* [ ] [PDF::Tags](https://modules.raku.org/dist/PDF::Tags) – Fail, Bisected: [84ce97d](https://github.com/rakudo/rakudo/commit/84ce97d255640f68e7f6300685ee0ab1c481ce2c)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: PDF::Tags
  ===> Found: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7> [via Zef::Repository::LocalCache]
  ===> Testing: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  [PDF::Tags] t/01basic.t .............. ok
  [PDF::Tags] t/02xpath-basic.t ........ ok
  [PDF::Tags] t/03xpath-predicates.t ... ok
  [PDF::Tags] t/04xpath-node-test.t .... ok
  [PDF::Tags] t/05xpath-axis.t ......... ok
  [PDF::Tags] t/06xml.t ................ ok
  [PDF::Tags] t/07copy-tree.t .......... ok
  [PDF::Tags] t/08-xpath-at-key-pos.t .. ok
  [PDF::Tags] t/write-tags.t ........... ok
  [PDF::Tags] All tests successful.
  [PDF::Tags] Files=9, Tests=104, 155 wallclock secs ( 0.03 usr  0.01 sys + 219.17 cusr 14.06 csys = 233.27 CPU)
  [PDF::Tags] Result: PASS
  [PDF::Tags] ...............
  ===> Testing [OK] for PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  ===> Installing: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  ===> Install [OK] for PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>

  1 bin/ script [pdf-tag-dump.raku] installed to:
  /tmp/2owebNReUv/bin

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: PDF::Tags
  ===> Found: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7> [via Zef::Repository::LocalCache]
  ===> Testing: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  [PDF::Tags] t/01basic.t .............. ok
  [PDF::Tags] t/02xpath-basic.t ........ ok
  [PDF::Tags] t/03xpath-predicates.t ... ok
  [PDF::Tags] t/04xpath-node-test.t .... ok
  [PDF::Tags] t/05xpath-axis.t ......... ok
  [PDF::Tags] t/06xml.t ................ ok
  [PDF::Tags] .............Type check failed in binding to parameter '<anon>'; expected Hash[Bool,Any] but got Hash[Bool,Any] ({})
  [PDF::Tags]   in method fetch at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 34
  [PDF::Tags]   in method Hash at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 63
  [PDF::Tags]   in method max-key at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 28
  [PDF::Tags]   in method mark at /mnt/data/zef-data/store/PDF-Tags-0.0.3.tar.gz/PDF-Tags-0.0.3/lib/PDF/Tags/Elem.rakumod (PDF::Tags::Elem) line 89
  [PDF::Tags]   in block <unit> at t/07copy-tree.t line 31
  [PDF::Tags] t/07copy-tree.t .......... 
  [PDF::Tags] Dubious, test returned 255 (wstat 65280, 0xff00)
  [PDF::Tags] Failed 3/3 subtests 
  [PDF::Tags] t/08-xpath-at-key-pos.t .. ok
  [PDF::Tags] Type check failed in binding to parameter '<anon>'; expected Hash[Bool,Any] but got Hash[Bool,Any] ({})
  [PDF::Tags]   in method fetch at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 34
  [PDF::Tags]   in method Hash at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 63
  [PDF::Tags]   in method max-key at /mnt/installed/PDF::Class_0.4.6/sources/26ABB1328C8D7E4826EB9AD3281BE960E5458E21 (PDF::NumberTree) line 28
  [PDF::Tags]   in method mark at /mnt/data/zef-data/store/PDF-Tags-0.0.3.tar.gz/PDF-Tags-0.0.3/lib/PDF/Tags/Elem.rakumod (PDF::Tags::Elem) line 89
  [PDF::Tags]   in block <unit> at t/write-tags.t line 31
  [PDF::Tags] t/write-tags.t ........... 
  [PDF::Tags] Dubious, test returned 255 (wstat 65280, 0xff00)
  [PDF::Tags] Failed 16/16 subtests 
  [PDF::Tags] Test Summary Report
  [PDF::Tags] -------------------
  [PDF::Tags] t/07copy-tree.t        (Wstat: 65280 Tests: 0 Failed: 0)
  [PDF::Tags]   Non-zero exit status: 255
  [PDF::Tags]   Parse errors: Bad plan.  You planned 3 tests but ran 0.
  [PDF::Tags] t/write-tags.t         (Wstat: 65280 Tests: 0 Failed: 0)
  [PDF::Tags]   Non-zero exit status: 255
  [PDF::Tags]   Parse errors: Bad plan.  You planned 16 tests but ran 0.
  [PDF::Tags] Files=9, Tests=85, 94 wallclock secs ( 0.03 usr  0.01 sys + 111.50 cusr  6.03 csys = 117.57 CPU)
  [PDF::Tags] Result: FAIL
  ===> Testing [FAIL]: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  [PDF::Tags] Failed to get passing tests, but continuing with --force-test
  ===> Installing: PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>
  ===> Install [OK] for PDF::Tags:ver<0.0.3>:auth<github:pdf-raku>:api<PDF.1.7>

  1 bin/ script [pdf-tag-dump.raku] installed to:
  /mnt/installed/PDF::Tags_0.0.3/bin

  ```
  </details>
* [ ] [Uzu](https://modules.raku.org/dist/Uzu) – Fail, Bisected: [84ce97d](https://github.com/rakudo/rakudo/commit/84ce97d255640f68e7f6300685ee0ab1c481ce2c)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: Uzu
  ===> Found: Uzu:ver<0.3.6> [via Zef::Repository::LocalCache]
  ===> Installing: Uzu:ver<0.3.6>
  ===> Install [OK] for Uzu:ver<0.3.6>

  1 bin/ script [uzu] installed to:
  /tmp/krPCdVYFB4/bin

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: Uzu
  ===SORRY!===
  STable Sub+{Callable[Str]} does not exist in serialization context

  ===> Found: Uzu:ver<0.3.6> [via Zef::Repository::LocalCache]
  ===> Installing: Uzu:ver<0.3.6>
  ===> Install [FAIL] for Uzu:ver<0.3.6>: ===SORRY!===
  STable Sub+{Callable[Str]} does not exist in serialization context


  ```
  </details>

| Status                    | Count |          Modules          |
| :------------------------ | :---: | :------------------------ |
| Flapper                   |     1 | [Net::DNS](https://modules.raku.org/dist/Net::DNS) |
| UnhandledException        |     1 | [Foo::�Foo](https://modules.raku.org/dist/Foo::�Foo) |
| InstallableButUntested    |     6 | [HTTP::Server::Async](https://modules.raku.org/dist/HTTP::Server::Async) [IO::Socket::Async::SSL](https://modules.raku.org/dist/IO::Socket::Async::SSL) [IRC::Client](https://modules.raku.org/dist/IRC::Client) [Log::Minimal](https://modules.raku.org/dist/Log::Minimal) [Time::Duration](https://modules.raku.org/dist/Time::Duration) [Toaster](https://modules.raku.org/dist/Toaster) |
| Fail                      |     7 | [AI::FANN](https://modules.raku.org/dist/AI::FANN) [Compress::Bzip2](https://modules.raku.org/dist/Compress::Bzip2) [MagickWand](https://modules.raku.org/dist/MagickWand) [[PDF::Tags](https://modules.raku.org/dist/PDF::Tags) [Uzu](https://modules.raku.org/dist/Uzu) |
| ZefFailure                |    10 | [Async::Workers](https://modules.raku.org/dist/Async::Workers) [Concurrent::BoundedChannel](https://modules.raku.org/dist/Concurrent::BoundedChannel) [DateTime::TimeZone](https://modules.raku.org/dist/DateTime::TimeZone) [File::Metadata::Libextractor](https://modules.raku.org/dist/File::Metadata::Libextractor) [Games::BubbleBreaker](https://modules.raku.org/dist/Games::BubbleBreaker) [MongoDB](https://modules.raku.org/dist/MongoDB) [Perl6::Ecosystem](https://modules.raku.org/dist/Perl6::Ecosystem) [SOAP::Client](https://modules.raku.org/dist/SOAP::Client) [Test::Time](https://modules.raku.org/dist/Test::Time) [WebService::Soundcloud](https://modules.raku.org/dist/WebService::Soundcloud) |
| MissingDependency         |    11 | [CSS::Module](https://modules.raku.org/dist/CSS::Module) [CSS::Selector::To::XPath](https://modules.raku.org/dist/CSS::Selector::To::XPath) [DBIish::Pool](https://modules.raku.org/dist/DBIish::Pool) [DBIx::NamedQueries](https://modules.raku.org/dist/DBIx::NamedQueries) [HTTP::Server::Middleware::JSON](https://modules.raku.org/dist/HTTP::Server::Middleware::JSON) [HTTP::Server::Router](https://modules.raku.org/dist/HTTP::Server::Router) [HTTP::Server::Router::YAML](https://modules.raku.org/dist/HTTP::Server::Router::YAML) [Hematite](https://modules.raku.org/dist/Hematite) [Task::Galaxy](https://modules.raku.org/dist/Task::Galaxy) [Termbox](https://modules.raku.org/dist/Termbox) [Web](https://modules.raku.org/dist/Web) |
| CyclicDependency          |    16 | [Algorithm::HierarchicalPAM](https://modules.raku.org/dist/Algorithm::HierarchicalPAM) [Algorithm::LDA](https://modules.raku.org/dist/Algorithm::LDA) [Algorithm::LibSVM](https://modules.raku.org/dist/Algorithm::LibSVM) [CSS](https://modules.raku.org/dist/CSS) [CSS::Properties](https://modules.raku.org/dist/CSS::Properties) [Chart::Gnuplot](https://modules.raku.org/dist/Chart::Gnuplot) [Cro::HTTP::Session::Red](https://modules.raku.org/dist/Cro::HTTP::Session::Red) [Foo::Dependencies::A-on-B](https://modules.raku.org/dist/Foo::Dependencies::A-on-B) [Foo::Dependencies::B-on-A](https://modules.raku.org/dist/Foo::Dependencies::B-on-A) [Foo::Dependencies::Self](https://modules.raku.org/dist/Foo::Dependencies::Self) [Geo::Hash](https://modules.raku.org/dist/Geo::Hash) [HTML::Canvas](https://modules.raku.org/dist/HTML::Canvas) [HTML::Canvas::To::PDF](https://modules.raku.org/dist/HTML::Canvas::To::PDF) [MeCab](https://modules.raku.org/dist/MeCab) [Red](https://modules.raku.org/dist/Red) [RedX::HashedPassword](https://modules.raku.org/dist/RedX::HashedPassword) |
| AlwaysFail                |   316 | ⋯                         |
| OK                        |  1173 | ⋯                         |

This run started on 2020-10-18T16:16:44Z and finished in ≈3 hours.

**Comment by Altai-man** [comment]
It seems the moarvm fix did cause regressions, plus NativeCall regressions.

**Comment by JJ** [comment]
The "non-zero wait status" is caused by that?

**Comment by lizmat** [comment]
That usually indicates some kind of exception, possibly a segfault

**Comment by lizmat** [comment]
Regarding `AI::FANN`, because of the `All 10 subtests passed`, it looks like it is crashing *after* having completed all tests.  Which points to an issue with `NativeCall` or `MoarVM` itself.  Which I'm afraid is really outside of my competency level  :-(

**Comment by Altai-man** [comment]
I suspect the problem with AI::FANN is the same as with Compress::Bzip2 (the error seems the same to me) and it's just that Blin bisected wrong commit? So at worst we can just revert the nativecall change.

**Comment by Altai-man** [comment]
Is reverting https://github.com/MoarVM/MoarVM/commit/b6d064a698c50842086f15aedd1aacc279086ba3 and https://github.com/rakudo/rakudo/commit/ecaf334c851de35c6e8fbca6a4012bcc4d21246c an option?

**Comment by lizmat** [comment]
@Altai-man did you notice I created a branch for https://github.com/rakudo/rakudo/commit/ecaf334c851de35c6e8fbca6a4012bcc4d21246c  for easier Blin testing ?

**Comment by niner** [comment]
On Donnerstag, 22. Oktober 2020 12:14:48 CEST Altai-man wrote:
> Is reverting
> https://github.com/MoarVM/MoarVM/commit/b6d064a698c50842086f15aedd1aacc2790
> 86ba3 and
> https://github.com/rakudo/rakudo/commit/ecaf334c851de35c6e8fbca6a4012bcc4d2
> 1246c an option?

Certainly! I have looked at the param interning issue and golfed down one 
issue somewhat, but have no idea about the time frame for a fix. This stuff is 
horribly complicated. Took me several weeks to come up with the initial fixes.

**Comment by lizmat** [comment]
@Altai-man let me know if I can help you with this.

**Comment by mid1221213** [comment]
My apologies for the problem with ecaf334... It passed the Rakudo test suite on NativeCall so I was thinking it was OK. I'm going to try to dig further to pinpoint the problem and hopefully create a new PR.

**Comment by lizmat** [comment]
> It passed the Rakudo test suite on NativeCall so I was thinking it was OK

@mid1221213 please don't worry about it.  That's when we learn we need to write more tests.  And that's why we have Blin to make sure we catch these types of problems in the wild before there is a release!

**Comment by Altai-man** [comment]
I am currently traveling until Saturday (morning is when I'm back, tired and all), I still expect to do a release given this issue is resolved. As fixes are complicated, I asked about revert possibility to make sure authors confirm it does not horribly break things in other ways I am not aware of.

Can't do Blin runs either now, but as it takes 3-5 hours making another run is not very critical.

**Comment by mid1221213** [comment]
> I suspect the problem with AI::FANN is the same as with Compress::Bzip2 (the error seems the same to me) and it's just that Blin bisected wrong commit? So at worst we can just revert the nativecall change.

I reverted locally my commit (ecaf334) and it seems that it is not related to the fail in `AI::FANN`.

I think I know where the problem in my commit is, and I managed to have a new one (actually I changed 2 lines in `NativeCall.rakumod` instead of only 1 that was really needed) that does not pose problem to `Compress::Bzip2` anymore. 

How should I submit it? A new PR with only the 1-line change, or a revert of my previous commit with a new one?
